src/main/java/org/opentripplanner/netex/mapping/FeedScopedIdFactory.java

```diff
@@ -0,0 +1,22 @@
+package org.opentripplanner.netex.mapping;
+
+import org.opentripplanner.model.FeedScopedId;
+
+/**
+ * GTFS uses the term agency_id, which is used to scope the ids of all other elements in OTP. Since NeTEx does not
+ * contain this id and also assumes ids are already unique, it is necessary to specify an id to use in the OTP model.
+ * <p>
+ * This factory is used to set the feed id once and then apply it to elements as they are created.
+ */
+public class FeedScopedIdFactory {
+    private static String feedId = "NETEX_AGENCY_ID_NOT_SET";
```

Comment:
Yes, I will update the comment - I also want to clean up the ID generation in OTP, making it a "per feed" or "instance" strategy (needs some thinking). This will replace the very simple "hack".

Reply:
This is related to #2524